synopsys-task/src/synopsys-task/tools-parameter.ts

```diff
@@ -313,19 +338,59 @@ export class SynopsysToolsParameter {
               password: inputs.COVERITY_USER_PASSWORD,
             },
             url: inputs.COVERITY_URL,
-            project: { name: inputs.COVERITY_PROJECT_NAME },
-            stream: { name: inputs.COVERITY_STREAM_NAME },
-          },
-          automation: {},
-          network: {
-            airGap: inputs.ENABLE_NETWORK_AIRGAP,
+            project: {},
+            stream: {},
           },
         },
-        project: {},
       },
     };
 
-    if (inputs.COVERITY_LOCAL) {
+    const azureRepositoryName = this.getAzureRepositoryName();
+
+    if (inputs.COVERITY_PROJECT_NAME) {
+      covData.data.coverity.connect.project.name = inputs.COVERITY_PROJECT_NAME;
+    } else {
+      taskLib.debug("COVERITY_PROJECT_NAME: " + azureRepositoryName);
+      covData.data.coverity.connect.project.name = azureRepositoryName;
+    }
+
+    if (inputs.COVERITY_STREAM_NAME) {
+      covData.data.coverity.connect.stream.name = inputs.COVERITY_STREAM_NAME;
+    } else {
+      if (isPullRequest()) {
+        const pullRequestTargetBranchName =
```

Comment:
can we try to simplify this like below for better readability, need to try once:

let streamName = inputs.COVERITY_STREAM_NAME;
if (!streamName) {
  const branchName = isPullRequest()
    ? taskLib.getVariable(AZURE_ENVIRONMENT_VARIABLES.AZURE_PULL_REQUEST_TARGET_BRANCH)
    : taskLib.getVariable(AZURE_ENVIRONMENT_VARIABLES.AZURE_SOURCE_BRANCH);
  streamName = azureRepositoryName && branchName
    ? azureRepositoryName.concat("-").concat(branchName)
    : "";
}
covData.data.coverity.connect.stream.name = streamName;

Reply:
Refactored